### ISSUE TITLE
Equality matcher behaviour is unexpected when comparing case classes that extend exception

### ISSUE THREAD

**Issue by mcintyre94** [etorreborre/specs2#866]
As Scala defines an `equals` method for case classes based on their data fields regardless of what they extend, I can do this in a Scala notebook:

```scala
case class AException(data: String) extends Exception

def somethingOrError: Either[AException, Int] = Left(AException("data"))

val b = somethingOrError

b == Left(AException("data")) // true
```

However in Specs2 the equality matcher seems to be ignoring the case class equality and comparing as a throwable:

```scala
package my.package

import org.specs2.mutable.Specification

class Cmtest extends Specification {
  case class AException(data: String) extends Exception

  def somethingOrError: Either[AException, Int] = Left(AException("data"))

  "exception equality" in {
    val b = somethingOrError

    println(b.equals(Left(AException("data"))))

    b ==== Left(AException("data"))
  }
}
```

This test fails with:

```
Left(
the stacktrace is incorrect

<large stack trace comparison>
```

I'd expect it to pass because despite being throwable, the comparators are equal by the case class equality. 

For what it's worth I spotted this trying to upgrade a project from 3.8.9 to the current latest, 4.10.3 - this test passed on 3.8.9

**Comment by etorreborre** [comment]
This is a bug introduced by the `Diffable` typeclass. I will fix it and publish a new version. Unfortunately there doesn't seem to be a work-around given the way implicits work (I'm curious to see if that would the same with Scala 3).

**Comment by mcintyre94** [comment]
Awesome, thanks for the quick fix! :)

**Comment by etorreborre** [comment]
The fix is available now in [`specs2-4.10.4`](https://repo1.maven.org/maven2/org/specs2/specs2-core_2.13/4.10.4/)

**Comment by SethTisue** [comment]
This broke some downstream projects in the community build: https://github.com/scala/community-build/pull/1254

(But perhaps the user code is at fault and the changes are as-intended?)

**Comment by etorreborre** [comment]
Thanks @SethTisue this seems to be an unexpected effect of that fix. I'm going to investigate those broken cases. 
BTW, it's a real pay-off to have a community build here because we can fix issues before they go too far!

**Comment by etorreborre** [comment]
It turns out that I have been away from Scala for too long. I forgot the difference between `==` and `equals`...

**Comment by etorreborre** [comment]
Released as [`specs2-4.10.5`](https://repo1.maven.org/maven2/org/specs2/specs2-core_2.13/4.10.5/
)
